tslint-config-5minds/tslint.json

```diff
@@ -1,39 +1,60 @@
 {
   "rules": {
     "adjacent-overload-signatures": true,
-    "member-access": [true, "check-accessor"],
-    "member-ordering": [
-      true, {
-        "order": [
-          "static-field",
-          "instance-field",
-          "constructor",
-          "static-method",
-          "instance-method"
-        ]
-      }
-    ],
+    "ban-ts-ignore": true,
+    "ban-types": {
+      "options": [
+        ["Object", "Use {} instead."],
+        ["String", "Use 'string' instead."],
+        ["Number", "Use 'number' instead."],
+        ["Boolean", "Use 'boolean' instead."]
+      ]
+    },
+    "member-access": {
+      "options": ["check-accessor"]
+    },
+    "member-ordering": {
+      "options": [
+        {
+          "order": [
+            "static-field",
+            "instance-field",
+            "constructor",
+            "static-method",
+            "instance-method"
+          ]
+        }
+      ]
+    },
     "no-any": true,
     "no-empty-interface": true,
+    "no-import-side-effect": {
+      "options": [{"ignore-module": "(\\.html|\\.css)$"}]
+    },
     "no-inferrable-types": false,
     "no-internal-module": true,
     "no-magic-numbers": true,
     "no-namespace": true,
+    "no-non-null-assertion": true,
+    "no-parameter-reassignment": true,
     "no-reference": true,
     "no-var-requires": true,
-    "only-arrow-functions": [true, "allow-named-functions"],
+    "only-arrow-functions": {
+      "options": ["allow-named-functions"]
+    },
     "prefer-for-of": true,
     "promise-function-async": false,
-    "typedef": [
-      true,
-      "call-signature",
-      "arrow-call-signature",
-      "parameter",
-      "arrow-parameter",
-      "property-declaration",
-      "variable-declaration",
-      "member-variable-declaration"
-    ],
+    "typedef": {
+      "options": [
+        "call-signature",
+        "arrow-call-signature",
+        "parameter",
+        "arrow-parameter",
+        "property-declaration",
+        "variable-declaration",
```

Comment:
Das ist vermutlich out-of-scope von diesem PR, aber ich habe mich immer gefragt, wieso wir bei TypeScript den Typ der Variable **immer** angeben müssen, obwohl der Compiler den Typ per [Type Inference](https://en.wikipedia.org/wiki/Type_inference) bestimmen kann. TypeScript kann das schon seit längerer Zeit (siehe auch https://www.typescriptlang.org/docs/handbook/type-inference.html).

In .NET Projekten erlauben (bzw. erzwingen!) wir, den Typ der Variable wegzulassen, wenn der Compiler den Typ durch die Angabe auf der rechten Seite selbstständig ermitteln kann. Warum also nicht in TypeScript Projekten?

(Eventuell sollte man aus diesem Kommentar ein Issue machen. Da gibt es sicherlich Diskussionsbedarf.)

Reply:
@NullEnt1ty Da bin ich voll und Ganz bei dir. 
Auch der Code Lesbarkeit kann das nur gut tun.